lxd/storage/drivers/driver_btrfs.go

```diff
@@ -304,23 +304,70 @@ func (d *btrfs) Validate(config map[string]string) error {
 func (d *btrfs) Update(changedConfig map[string]string) error {
 	// We only care about btrfs.mount_options.
 	val, ok := changedConfig["btrfs.mount_options"]
-	if !ok {
-		return nil
-	}
+	if ok {
+		// Custom mount options don't work inside containers
+		if d.state.OS.RunningInUserNS {
+			return nil
+		}
 
-	// Custom mount options don't work inside containers
-	if d.state.OS.RunningInUserNS {
-		return nil
+		// Trigger a re-mount.
+		d.config["btrfs.mount_options"] = val
+		mntFlags, mntOptions := filesystem.ResolveMountOptions(strings.Split(d.getMountOptions(), ","))
+		mntFlags |= unix.MS_REMOUNT
+
+		err := TryMount("", GetPoolMountPath(d.name), "none", mntFlags, mntOptions)
+		if err != nil {
+			return err
+		}
 	}
 
-	// Trigger a re-mount.
-	d.config["btrfs.mount_options"] = val
-	mntFlags, mntOptions := filesystem.ResolveMountOptions(strings.Split(d.getMountOptions(), ","))
-	mntFlags |= unix.MS_REMOUNT
+	size, ok := changedConfig["size"]
+	if ok {
+		// Figure out loop path
+		loopPath := loopFilePath(d.name)
 
-	err := TryMount("", GetPoolMountPath(d.name), "none", mntFlags, mntOptions)
-	if err != nil {
-		return err
+		if filepath.IsAbs(d.config["source"]) {
+			loopPath = d.config["source"]
+		}
+
+		// Resize loop file
+		f, err := os.OpenFile(loopPath, os.O_RDWR, 0600)
+		if err != nil {
+			return err
+		}
+
+		defer func() { _ = f.Close() }()
+
+		sizeBytes, _ := units.ParseByteSizeString(size)
+
+		err = unix.Fallocate(int(f.Fd()), 0, 0, sizeBytes)
+		if err != nil {
+			return err
+		}
+
+		loopDevPath, err := loopDeviceSetup(loopPath)
+		if err != nil {
+			return err
+		}
+
+		defer func() { _ = loopDeviceAutoDetach(loopDevPath) }()
+
+		err = loopDeviceSetCapacity(loopDevPath)
+		if err != nil {
+			return err
+		}
+
+		_, err = d.Mount()
```

Comment:
Shouldnt the storage pool already be mounted if LXD is running, and why is it then unmounted?
What if instances are running?

Reply:
I'll be dropping this part as there's no need to mount the pool.

Reply:
Also, resizing works when instances are running.